api/http/proxy/factory/docker/volumes.go

```diff
@@ -122,7 +129,50 @@ func (transport *Transport) decorateVolumeResourceCreationOperation(request *htt
 	}
 
 	if response.StatusCode == http.StatusCreated {
-		err = transport.decorateGenericResourceCreationResponse(response, resourceIdentifierAttribute, resourceType, tokenData.ID)
+		err = transport.decorateVolumeCreationResponse(response, resourceIdentifierAttribute, resourceType, tokenData.ID)
 	}
 	return response, err
 }
+
+func (transport *Transport) decorateVolumeCreationResponse(response *http.Response, resourceIdentifierAttribute string, resourceType portainer.ResourceControlType, userID portainer.UserID) error {
+	responseObject, err := responseutils.GetResponseAsJSONOBject(response)
+	if err != nil {
+		return err
+	}
+
+	if responseObject["Name"] == nil {
+		log.Printf("[ERROR] [proxy,docker]")
+		return errors.New("missing identifier in Docker resource creation response")
+	}
+
+	resourceID := responseObject["Name"].(string) + responseObject["CreatedAt"].(string)
+
+	resourceControl, err := transport.createPrivateResourceControl(resourceID, resourceType, userID)
+	if err != nil {
+		return err
+	}
+
+	responseObject = decorateObject(responseObject, resourceControl)
+
+	return responseutils.RewriteResponse(response, responseObject, http.StatusOK)
+}
+
+func (transport *Transport) volumeDetailOperation(requestPath string, request *http.Request) (*http.Response, error) {
+
+	if request.Method == http.MethodGet {
+		return transport.rewriteOperation(request, transport.volumeInspectOperation)
+	}
+
+	cli := transport.dockerClient
+	volume, err := cli.VolumeInspect(context.Background(), path.Base(requestPath))
+	if err != nil {
+		return nil, err
+	}
+
+	volumeID := volume.Name + volume.CreatedAt
+
+	if request.Method == http.MethodDelete {
+		return transport.executeGenericResourceDeletionOperation(request, volumeID, portainer.VolumeResourceControl)
```

Comment:
I think that you'll need to introduce an override method for the `executeGenericResourceDeletionOperation`. Something like `executeVolumeResourceDeletionOperation`.

You'll need to pass the volume name and the volume ID, as you want Docker to remove the resource based on the volume name and Portainer to remove the resource control based on the volume ID.